src/Resources/Completions.php

```diff
@@ -18,13 +19,24 @@ final class Completions
      *
      * @param  array<string, mixed>  $parameters
      */
-    public function create(array $parameters): CreateResponse
+    public function create(array $parameters): CreateResponse|Generator
     {
         $payload = Payload::create('completions', $parameters);
 
-        /** @var array{id: string, object: string, created: int, model: string, choices: array<int, array{text: string, index: int, logprobs: array{tokens: array<int, string>, token_logprobs: array<int, float>, top_logprobs: array<int, string>|null, text_offset: array<int, int>}|null, finish_reason: string}>, usage: array{prompt_tokens: int, completion_tokens: int, total_tokens: int}} $result */
-        $result = $this->transporter->requestObject($payload);
+        /** @var array{id: string, object: string, created: int, model: string, choices: array<int, array{text: string, index: int, logprobs: array{tokens: array<int, string>, token_logprobs: array<int, float>, top_logprobs: array<int, string>|null, text_offset: array<int, int>}|null, finish_reason: string}>, usage: array{prompt_tokens: int, completion_tokens: int, total_tokens: int}}|Generator $result */
+        $result = $this->transporter->requestObject($payload, $parameters['stream'] ?? false);
 
-        return CreateResponse::from($result);
+        if (is_array($result)) {
+            return CreateResponse::from($result);
+        }
+
+        return $this->stream($result);
+    }
+
+    private function stream(Generator $stream): Generator
+    {
+        foreach ($stream as $data) {
```

Comment:
Can't you create a single `CreateResponse` object, that behind the scenes allows to iterate on the stream?

Reply:
`CreateResponse` for completions now implements `Iterator` interface which allows to iterate over the stream. How do you feel about that?